clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1274,19 +1275,19 @@ def GlobalOp : CIR_Op<"global", [Symbol]> {
                        OptionalAttr<AnyAttr>:$initial_value,
                        UnitAttr:$constant,
                        OptionalAttr<I64Attr>:$alignment);
-
+  let regions = (region AnyRegion:$ctorRegion, AnyRegion:$dtorRegion);
   let assemblyFormat = [{
        ($sym_visibility^)?
        (`constant` $constant^)?
        $linkage
        $sym_name
-       custom<GlobalOpTypeAndInitialValue>($sym_type, $initial_value)
+       custom<GlobalOpTypeAndInitialValue>($sym_type, $initial_value, $ctorRegion, $dtorRegion)
        attr-dict
   }];
 
   let extraClassDeclaration = [{
     bool isDeclaration() {
-      return !getInitialValue();
+      return !getInitialValue() && getCtorRegion().empty();
```

Comment:
You did introduce dtor stuff in GlobalOp already though, so it's a bit odd that you neither add it here nor add a FIXME or something - there's a chance this could slip by when dtors get introduced.

I'm all for incremental work, but if what is being added is not exercised I'd rather see it coming as a follow up work in another PR. For this patch I understand you're not adding codegen support for dtor, but expect that what already got added is somehow exercised (parsing/printing tests is an option).